### PR TITLE
feat(*): shipping options update API

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- Update the `updateShippingOptions` module service API to follow the newest convention
- Add upsert support for shipping options
- update shipping options workflow
- update shipping options api end point and validation
- update shipping options all integration tests

FIXES CORE-1926

### CODE REVIEW

File: packages/fulfillment/src/services/fulfillment-module-service.ts [hunk 1]
```diff
@@ -1079,6 +1097,82 @@ export default class FulfillmentModuleService<
...
+    )
+
+    const allShippingOptions = await this.baseRepository_.serialize<
+      FulfillmentTypes.ShippingOptionDTO[] | FulfillmentTypes.ShippingOptionDTO
+    >(upsertedShippingOptions)
+
+    return Array.isArray(data) ? allShippingOptions : allShippingOptions[0]
+  }
+
+  @InjectManager("baseRepository_")
```
Comment (olivermrbl) [inline packages/fulfillment/src/services/fulfillment-module-service.ts]:
```suggestion
  @InjectTransactionManager("baseRepository_")
```

**Review by olivermrbl — APPROVED** [review]
Nice work!

File: packages/fulfillment/src/services/fulfillment-module-service.ts (hunk 1 above)
Comment (adrien2p) [reply packages/fulfillment/src/services/fulfillment-module-service.ts]: ahah I did fix it in the other pr 😂

Comment (adrien2p) [reply packages/fulfillment/src/services/fulfillment-module-service.ts]: let me move it

**Review by fPolic — APPROVED** [review]

**Pull request merged**